bdk_chain/src/chain_graph.rs

```diff
@@ -52,25 +52,6 @@ impl<P: ChainPosition> ChainGraph<P> {
         self.chain.set_checkpoint_limit(limit)
     }
 
-    /// Inserts a transaction into the inner [`ChainGraph`] and optionally into the chain at
-    /// `position`.
-    ///
-    /// **Warning**: This function modifies the internal state of the chain graph. You are
-    /// responsible for persisting these changes to disk if you need to restore them.
-    pub fn insert_tx(
-        &mut self,
-        tx: Transaction,
-        position: Option<P>,
-    ) -> Result<bool, sparse_chain::InsertTxErr> {
-        let chain_changed = match position {
-            Some(pos) => self.chain.insert_tx(tx.txid(), pos)?,
-            None => false,
-        };
-        let graph_changed = self.graph.insert_tx(tx);
-
-        Ok(graph_changed || chain_changed)
-    }
-
```

Comment:
I think we should not delete this API. Only giving `insert_tx_changeset` means the user has to deal with an error case when generating it AND applying it. I think this policy is better:

1. Have a function which returns the changeset with non mut `self`.
2. Have a utility function that genrates **and** applies the changeset and returns the changeset (by cloneing). This one would be called `insert_tx` for example.
3. Have a function that applies the changeset.
4. Make all changeset types `[must_use]`.

Let's apply this policy to all methods on all changeset types.